### PR TITLE
fix: remove decoder service

### PR DESCRIPTION
**Pull request by amanPras**
## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test no regressions introduced

## Checklist:
- [x] Each of my commits contains one meaningful change
- [x] I have performed rebase of my branch on top of the development
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by ap891843** [comment]
PR is draft as testing in progress

**Comment by slavek-kucera** [comment]
With this change the server relies on the lsp client to send normalized URIs. Will that be a problem? Should the incoming URIs be normalized e.g. in the LSP message handlers?

**Comment by amanPras** [comment]
> With this change the server relies on the lsp client to send normalized URIs. Will that be a problem? Should the incoming URIs be normalized e.g. in the LSP message handlers?

Yes. This was the  initial idea. 
First step was to do some clean up - get rid of  UriDecodeService.  It was poorly implemented and was breaking a lot of cases

Then decide the appropriate place to normalize the URI's. I am still not sure, what would be the correct place for it. Should it sit on client or the LS server ? 
For example:

- custom URI object on LS <-- this would help us overcome some of the java limitation, but could be extensive
-  LS middle ware client <-- Doubtful as it break one server , multi client 
- LSP message handlers , as you suggested

**Comment by ishche** [comment]
> * LSP message handlers , as you suggested

By LSP values we should minimize client specific changes. I like idea with custom URI class, for now it can be wrapper on string with cleanup constructor.

**Comment by amanPras** [comment]
> With this change the server relies on the lsp client to send normalized URIs. Will that be a problem? Should the incoming URIs be normalized e.g. in the LSP message handlers?

fyi, tested it on shared drive ( windows shared drive using hostname in URI and basic testing on mac and win). Seems to work. But fails for  path like //./c:, which is a known issue

**Comment by slavek-kucera** [comment]
I'd also prefer a special type (value class) that would contain the normalized uri - I expect a lot of changes in the rest of the server though.

**Review by Nurkambay — DISMISSED** [review]

**Comment by Nurkambay** [comment]
Let's add unit tests first if we can break something with this PR. For example, if we suppose that the Client will send us a normalized URI, let's add unit tests that are failing if received URI is not normalized

File: server/engine/src/test/java/org/eclipse/lsp/cobol/service/utils/WorkspaceFileServiceTest.java [hunk 1]
```diff
@@ -148,4 +150,25 @@ void getNameFromURIWithQueryDataset() {
             .getNameFromURI(
                 "file:///c:/workspace/POSITIVE_TESTS/.e4e/AD1DEV.PUBLIC.COTPA01.COBOL(TSTJUST).cbl?%7B%22service%22:%7B%22credential%22:%7B%22"));
   }
+
+  @Test
+  void getPathFromURI() {
+    WorkspaceFileService workspaceFileService = new WorkspaceFileService();
+    String unNormalizedURI = "file:///c:/workspace/path/wi th/spe&cial!@#chars";
+    assertThrows(
+        IllegalArgumentException.class, () -> workspaceFileService.getPathFromURI(unNormalizedURI));
```
Comment (amanPras) [inline server/engine/src/test/java/org/eclipse/lsp/cobol/service/utils/WorkspaceFileServiceTest.java:159]: @Nurkambay  added this as an example

**Comment by amanPras** [comment]
> Let's add unit tests first if we can break something with this PR. For example, if we suppose that the Client will send us a normalized URI, let's add unit tests that are failing if received URI is not normalized

updated

File: server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/TestFileDescriptor.java [hunk 2]
```diff
@@ -205,10 +204,10 @@ void test() {
   }
 
   private void assertHover(AnalysisResult result) {
-    UriDecodeService uriDecodeService = mock(UriDecodeService.class);
-    when(uriDecodeService.decode(DOCUMENT_URI)).thenReturn(DOCUMENT_URI);
+//    UriDecodeService uriDecodeService = mock(UriDecodeService.class);
```
Comment (Nurkambay) [inline server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/TestFileDescriptor.java]: Do we need these lines?

**Review by Nurkambay — APPROVED** [review]

File: server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/TestFileDescriptor.java (hunk 2 above)
Comment (amanPras) [reply server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/TestFileDescriptor.java]: Agree.
removed commented out code at e1e6fdb58aa6bea9f9708c9d1a505c81d544b93d

**Review by ishche — APPROVED** [review]

**Pull request merged**